### ISSUE TITLE
Enable clustering algorithms to select models on the validation set

### ISSUE THREAD

**Issue by WenjieDu** [WenjieDu/PyPOTS#203]
### 1. Feature description

Make the clustering algorithms to select the best model on the validation set during training.

### 2. Motivation

Currently we select the best clustering model according to the training loss but this may lead to overfitting. Although clustering models are unsupervised, users may want to select the best model on a given validation set during training.

### 3. Your contribution

Will make a PR to accomplish this.
